src/inventory/weapons/InstantWeapon.cpp

```diff
@@ -36,6 +37,20 @@ void InstantWeapon::fire(Marine &marine){
     //get all targets in line with the shot
     std::priority_queue<const HitBox*> targets;
     targets = collisionHandler.detectLineCollision(marine, getRange());
+   
+
+    //similar as to what happens in the line collision and is used to show a use of the line effect
+    const double degrees = marine.getAngle() - 90;
+    const double radians = degrees * M_PI / 180;
+    const int playerX = marine.getX() + (MARINE_WIDTH / 2);
+    const int playerY = marine.getY() + (MARINE_HEIGHT / 2);
+    const int deltaX  = playerX + range * cos(radians);
+    const int deltaY  = playerY + range * sin(radians);
+
+    //5 frames to display ie 1/12th of a second at 60fps
+    //0 red, 255 green, 0 blue
+    VisualEffect::instance().addPreLine(5, playerX, playerY, deltaX, deltaY, 0, 255, 0);
```

Comment:
I didnt know what we would want as an effect so i added this to demo it by making a green line as the laser shot fired